diff --git a/internal/common/os_name.bzl b/internal/common/os_name.bzl
index f45cfe5..70c0e51 100755
--- a/internal/common/os_name.bzl
+++ b/internal/common/os_name.bzl
@@ -20,6 +20,7 @@ OS_ARCH_NAMES = [
     ("windows", "amd64"),
     ("linux", "amd64"),
     ("linux", "arm64"),
+    ("linux", "armv7"),
 ]
 
 OS_NAMES = ["_".join(os_arch_name) for os_arch_name in OS_ARCH_NAMES]
@@ -43,6 +44,8 @@ def os_name(rctx):
         arch = rctx.execute(["uname", "-m"]).stdout.strip()
         if arch == "aarch64":
             return OS_NAMES[3]
+        if arch == "armv7l":
+            return OS_NAMES[4]
         return OS_NAMES[2]
     else:
         fail("Unsupported operating system: " + os_name)
@@ -55,4 +58,4 @@ def is_windows_os(rctx):
 
 def is_linux_os(rctx):
     name = os_name(rctx)
-    return name == OS_NAMES[2] or name == OS_NAMES[3]
+    return name == OS_NAMES[2] or name == OS_NAMES[3] or name == OS_NAMES[4]
diff --git a/internal/node/launcher.sh b/internal/node/launcher.sh
index e685da7..955d8b9 100755
--- a/internal/node/launcher.sh
+++ b/internal/node/launcher.sh
@@ -130,6 +130,7 @@ else
     *)
       case "${unameArch}" in
         aarch64*) readonly node_toolchain="nodejs_linux_arm64/bin/nodejs/bin/node" ;;
+        armv7l*) readonly node_toolchain="nodejs_linux_armv7/bin/nodejs/bin/node" ;;
         *) readonly node_toolchain="nodejs_linux_amd64/bin/nodejs/bin/node" ;;
       esac
       ;;
diff --git a/internal/node/node_repositories.bzl b/internal/node/node_repositories.bzl
index 5143ccb..e022297 100755
--- a/internal/node/node_repositories.bzl
+++ b/internal/node/node_repositories.bzl
@@ -222,6 +222,7 @@ BUILT_IN_NODE_PLATFORMS = [
     "darwin_amd64",
     "linux_amd64",
     "linux_arm64",
+    "linux_armv7",
     "windows_amd64",
 ]
 
diff --git a/internal/node/node_versions.bzl b/internal/node/node_versions.bzl
index c3ca42b..06bbdcd 100755
--- a/internal/node/node_versions.bzl
+++ b/internal/node/node_versions.bzl
@@ -622,6 +622,7 @@ NODE_VERSIONS = {
     # 12.13.0
     "12.13.0-darwin_amd64": ("node-v12.13.0-darwin-x64.tar.gz", "node-v12.13.0-darwin-x64", "49a7374670a111b033ce16611b20fd1aafd3296bbc662b184fe8fb26a29c22cc"),
     "12.13.0-linux_arm64": ("node-v12.13.0-linux-arm64.tar.xz", "node-v12.13.0-linux-arm64", "d65b3ce27639f15ae22941e3ff98a1c900aa9049fcc15518038615b0676037d5"),
+    "12.13.0-linux_armv7": ("node-v12.13.0-linux-armv7l.tar.gz", "node-v12.13.0-linux-armv7l", "c8bb1fca0712f360eaeeeab064426f8fb6f9af50144658aa1b50c9703fc7f680"),
     "12.13.0-linux_amd64": ("node-v12.13.0-linux-x64.tar.xz", "node-v12.13.0-linux-x64", "7a57ef2cb3036d7eacd50ae7ba07245a28336a93652641c065f747adb2a356d9"),
     "12.13.0-windows_amd64": ("node-v12.13.0-win-x64.zip", "node-v12.13.0-win-x64", "6f920cebeecb4957b4ef0def6d9b04c49d4582864f8d1a207ce8d0665865781a"),
     # 12.13.1
diff --git a/toolchains/node/BUILD.bazel b/toolchains/node/BUILD.bazel
index 02a0559..b035159 100755
--- a/toolchains/node/BUILD.bazel
+++ b/toolchains/node/BUILD.bazel
@@ -43,6 +43,14 @@ platform(
     ],
 )
 
+platform(
+    name = "linux_armv7",
+    constraint_values = [
+        "@bazel_tools//platforms:linux",
+        "@bazel_tools//platforms:arm",
+    ],
+)
+
 platform(
     name = "windows_amd64",
     constraint_values = [
@@ -74,6 +82,7 @@ alias(
         "@bazel_tools//src/conditions:darwin": "@nodejs_darwin_amd64_config//:toolchain",
         "@bazel_tools//src/conditions:darwin_x86_64": "@nodejs_darwin_amd64_config//:toolchain",
         "@bazel_tools//src/conditions:linux_aarch64": "@nodejs_linux_arm64_config//:toolchain",
+        "@bazel_tools//src/conditions:linux_armv7": "@nodejs_linux_armv7_config//:toolchain",
         "@bazel_tools//src/conditions:linux_x86_64": "@nodejs_linux_amd64_config//:toolchain",
         "@bazel_tools//src/conditions:windows": "@nodejs_windows_amd64_config//:toolchain",
         "//conditions:default": "@nodejs_linux_amd64_config//:toolchain",
@@ -88,6 +97,7 @@ alias(
         "@bazel_tools//src/conditions:darwin": "@nodejs_darwin_amd64//:node_bin",
         "@bazel_tools//src/conditions:darwin_x86_64": "@nodejs_darwin_amd64//:node_bin",
         "@bazel_tools//src/conditions:linux_aarch64": "@nodejs_linux_arm64//:node_bin",
+        "@bazel_tools//src/conditions:linux_armv7": "@nodejs_linux_armv7//:node_bin",
         "@bazel_tools//src/conditions:linux_x86_64": "@nodejs_linux_amd64//:node_bin",
         "@bazel_tools//src/conditions:windows": "@nodejs_windows_amd64//:node_bin",
         "//conditions:default": "@nodejs_linux_amd64//:node_bin",
@@ -115,6 +125,16 @@ toolchain(
     toolchain_type = ":toolchain_type",
 )
 
+toolchain(
+    name = "node_linux_armv7_toolchain",
+    target_compatible_with = [
+        "@bazel_tools//platforms:linux",
+        "@bazel_tools//platforms:armv7",
+    ],
+    toolchain = "@nodejs_linux_armv7_config//:toolchain",
+    toolchain_type = ":toolchain_type",
+)
+
 toolchain(
     name = "node_darwin_amd64_toolchain",
     target_compatible_with = [
